lnp2p/src/legacy/types.rs

```diff
@@ -744,6 +746,55 @@ impl LightningDecode for AddressList {
     }
 }
 
+#[derive(
+    Display, Copy, Clone, Debug, PartialEq, Hash, Eq, PartialOrd, Ord, Wrapper,
+)]
+#[display("{0} sec")]
+pub struct Timestamp(u32);
+
+impl<T> From<DateTime<T>> for Timestamp
+where
+    T: TimeZone,
+{
+    fn from(dt: DateTime<T>) -> Timestamp {
+        Timestamp(dt.timestamp() as u32)
+    }
+}
+
+impl From<u32> for Timestamp {
+    fn from(seconds: u32) -> Timestamp {
+        Timestamp(seconds)
+    }
+}
+
+impl Default for Timestamp {
+    fn default() -> Self {
+        Timestamp(Utc::now().timestamp() as u32)
+    }
+}
+
+impl From<Timestamp> for DateTime<Utc> {
+    fn from(ts: Timestamp) -> DateTime<Utc> {
+        Utc.timestamp(ts.0 as i64, 0)
+    }
+}
+
+impl From<Timestamp> for DateTime<Local> {
+    fn from(ts: Timestamp) -> DateTime<Local> {
+        Local.timestamp(ts.0 as i64, 0)
+    }
+}
+
+impl Timestamp {
+    pub fn checked_add(self, rhs: Timestamp) -> Option<Timestamp> {
+        Timestamp(self.0.checked_add(rhs.0).unwrap()).into()
```

Comment:
```suggestion
       self.0.checked_add(rhs.0).map(Timestamp::from)
```

Reply:
What do think about mapping the whole type like this? I checked and it returns None in case of overflow

self.0.checked_add(rhs.0).map(Timestamp)